TestRegisterSEESubscriptionReconnectsStreamOnError fails on windows

```
--- FAIL: TestRegisterSEESubscriptionReconnectsStreamOnError (0.50s)
        Error Trace:    subscription_test.go:429
        Error:          did not receive event in time
```

I am able to reproduce it only on Windows (tested golang 1.8.3 and 1.6.2, Windows 10 Pro Creators Update).  Can be solved by setting http client timeout to e.g. 50 milliseconds -  PR ok?

Also notice that timeout must be very low < 100 ms. Otherwise it fails even when SSEConnectWaitTime or eventPublishTimeout is set to seconds)

Logs during registerSSESubscription (with custom log line)
```
Error subscribing to Marathon event stream: Get http://127.0.0.1:63307/v2/events: dial tcp 127.0.0.1:63307: i/o timeout
Error subscribing to Marathon event stream: Get http://127.0.0.1:63307/v2/events: dial tcp 127.0.0.1:63307: i/o timeout
Error subscribing to Marathon event stream: Get http://127.0.0.1:63307/v2/events: dial tcp 127.0.0.1:63307: i/o timeout
Listen on SSE subscription 2017-07-06 12:43:58.124575 +0000 UTC
Error on SSE subscription: EOF, 2017-07-06 12:44:01.6705144 +0000 UTC
--- FAIL: TestRegisterSEESubscriptionReconnectsStreamOnError (4.00s)
```
Hangs for 3 seconds inside eventsource.SubscribeWith call

--- a/subscription_test.go
+++ b/subscription_test.go
@@ -20,9 +20,11 @@ import (
 	"testing"
 	"time"
 
+	"net"
+	"net/http"
+
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
-	"net/http"
 )
 
 const (
@@ -393,10 +395,14 @@ func TestConnectToSSEFailure(t *testing.T) {
 
 func TestRegisterSEESubscriptionReconnectsStreamOnError(t *testing.T) {
 	clientCfg := NewDefaultConfig()
-	clientCfg.HTTPClient = &http.Client{
-		// set timeout to small fraction of SSEConnectWaitTime to give client enough time
-		// to detect error and reconnect during sleep
-		Timeout: SSEConnectWaitTime / 10,
+	clientCfg.HTTPSSEClient = &http.Client{
+		Transport: &http.Transport{
+			Dial: (&net.Dialer{
+				// set timeout to a small fraction of SSEConnectWaitTime to give client enough time
+				// to detect error and reconnect during sleep
+				Timeout: SSEConnectWaitTime / 10,
+			}).Dial,
+		},
 	}
 	clientCfg.EventsTransport = EventsTransportSSE
 	config := configContainer{client: &clientCfg}

--- a/subscription_test.go
+++ b/subscription_test.go
@@ -17,11 +17,10 @@ limitations under the License.
 package marathon
 
 import (
-	"testing"
-	"time"
-
 	"net"
 	"net/http"
+	"testing"
+	"time"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"

--- a/subscription_test.go
+++ b/subscription_test.go
@@ -22,6 +22,7 @@ import (
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	"net/http"
 )
 
 const (
@@ -392,6 +393,11 @@ func TestConnectToSSEFailure(t *testing.T) {
 
 func TestRegisterSEESubscriptionReconnectsStreamOnError(t *testing.T) {
 	clientCfg := NewDefaultConfig()
+	clientCfg.HTTPClient = &http.Client{
+		// set timeout to small fraction of SSEConnectWaitTime to give client enough time
+		// to detect error and reconnect during sleep
+		Timeout: SSEConnectWaitTime / 10,
+	}
 	clientCfg.EventsTransport = EventsTransportSSE
 	config := configContainer{client: &clientCfg}
 

